src/cgutils.cpp

```diff
@@ -330,6 +330,11 @@ static unsigned julia_alignment(jl_value_t *jt)
         // Array always has this alignment
         return JL_SMALL_BYTE_ALIGNMENT;
     }
+    if (jt == (jl_value_t*)jl_datatype_type) {
```

Comment:
```suggestion
    if (jl_is_mutable(jt)) {
```

Assuming llvm-alloc-opt is willing to commit to this also?

Reply:
I don't think that's currently the case even for heap allocation. `Ref{Int8}()` on 32bit arch is only 8 byte aligned.

For stack allocation, I don't want to force any strong alignment guarantee than necessary (especially since 16byte alignment on entry isn't guaranteed everywhere). I assume it'll do a good job if aligning the allocation is benefitial. Cases that are handled in llvm-alloc-opt should be fairly easy for llvm to reason about.

For known heap address, we could certainly let llvm know when better alignment is available. However, that's also not what this function is used for. I'm pretty sure in general this function is also used for alignment of inline storage, where we certainly don't always have 16 byte of alignment.